### PR TITLE
Fix grid

### PR DESCRIPTION
**Pull request by davidgarry**
Fix grid proptypes
Fix width and margin container

### CODE REVIEW

File: src/elements/Grid/index.js [hunk 1]
```diff
@@ -67,49 +67,49 @@ export const getOffset = (colNumber, size) => {
...
 
+export const getContainerMarginString = size => {
+  if (grid[size].gutter === 'auto') {
+    return `margin: 0 auto;`;
+  }
+  return `margin: 0 ${grid[size].gutter + grid[size].type};`;
+};
+
+export const getContainerWidthString = size => {
+  if (Object.prototype.hasOwnProperty.call(grid[size], 'width')) {
```
Comment (bdo) [inline src/elements/Grid/index.js]: [KISS](https://en.wikipedia.org/wiki/KISS_principle)©
```suggestion
  if (grid[size].width)) {
```

File: src/elements/Grid/index.js [hunk 2]
```diff
@@ -67,49 +67,49 @@ export const getOffset = (colNumber, size) => {
...
+    return `width: ${grid[size].width + grid[size].type};`;
+  }
+  return '';
+};
+
 export const Container = styled.div`
-  margin: 0 ${grid.tiny.gutter + grid.tiny.type};
+  ${getContainerMarginString('tiny')};
```
Comment (bdo) [inline src/elements/Grid/index.js]: Maybe group them together into a get `getContainerStyles`

File: src/elements/Grid/index.stories.js [hunk 3]
```diff
@@ -37,7 +37,9 @@ gridStories
           medium={select('medium', ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'], '6')}
           large={select('large', ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'], '6')}
           wide={select('wide', ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12'], '3')}
-          mediumOffset={select('medium offset', ['1', '2', '3', '4', '5', '6'], '0')}
+          tinyOffset={select('medium offset', ['1', '2', '3', '4', '5', '6'], '0')}
```
Comment (bdo) [inline src/elements/Grid/index.stories.js]:
```suggestion
          tinyOffset={select('tiny offset', ['1', '2', '3', '4', '5', '6'], '0')}
```

File: src/elements/Grid/index.stories.js [hunk 4]
```diff
@@ -37,7 +37,9 @@ gridStories
...
+          smallOffset={select('medium offset', ['1', '2', '3', '4', '5', '6'], '0')}
```
Comment (bdo) [inline src/elements/Grid/index.stories.js]:
```suggestion
          smallOffset={select('small offset', ['1', '2', '3', '4', '5', '6'], '0')}
```

File: src/elements/Grid/index.js [hunk 5]
```diff
@@ -163,6 +163,8 @@ Column.propTypes = {
   medium: PropTypes.oneOf(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12']),
```
Comment (bdo) [inline src/elements/Grid/index.js]: Please create a couple of constants such as: `TWELVE_COLUMNS`, `SIX_COLUMNS` then we don't have to repeat ourselves. :wink:

**Comment by davidgarry** [comment]
We will merge this PR as @ldabiralai has reviewed it with @bdo and @leozero !

**Pull request merged**